src/Data/Form.purs

```diff
@@ -246,7 +251,7 @@ getValue
   -> Maybe a
 getValue x = userInput x.value <|> x.default
 
-setValue :: Key -> ExprType -> Page Expr -> Page Expr
+setValue :: Key -> InputSource ExprType -> Page Expr -> Page Expr
```

Comment:
This looks exactly the same as the work I did, except I made this `InputSourc Expr` instead of `ExprType`...looks like yours takes advantage of `InputSource` being a functor to wrap the value with `Val`, if there as a value. I like it, down to keep your way, it's more ergonomic

Reply:
Yeah, I felt pretty good that we had nearly identical solutions. It affirmed that I hadn't done the wrong thing 😄.